CenterForOpenScience/ember-osf-web
# [EMB-302][EMB-227] Modal to verify "add email"/"merge user" requests

## Purpose
Display a modal to confirm/verify requests to add an email or merge user accounts.


## Summary of Changes
- Update `osf-adapter` to use links from the API to create/update/delete, instead of assuming it knows how to build them
  - Allow specifying a `parentRelationship` for each adapter, if new records should be created at a related record's relationship endpoint
  - e.g. create emails at `/v2/users/me/settings/emails/`
- Add `user-email` model/adapter/serializer
- Add verification modal for new emails

![screen shot 2018-10-15 at 10 49 38](https://user-images.githubusercontent.com/6776190/46959177-6a65cc80-d069-11e8-8ea7-85a2e67c4070.png)
![screen shot 2018-10-15 at 13 41 27](https://user-images.githubusercontent.com/6776190/46967975-1a463480-d080-11e8-8414-a8604f32adbe.png)

## Side Effects
n/a


## Feature Flags
`ember_verify_email_modals` - Since this PR depends on not-yet-released API, the modals won't show unless this flag is enabled.


## QA Notes
- Add an unconfirmed email at `/settings/account`
- Click the link in the confirmation email, login
- You should be redirected to the dashboard and prompted to confirm the request. There shouldn't be a way to close the modal without selecting yes or no.


## Ticket


https://openscience.atlassian.net/browse/EMB-302
https://openscience.atlassian.net/browse/EMB-227

# Reviewer Checklist
